### PR TITLE
Utilities: Add jq-like json query functionality to `jp`

### PR DESCRIPTION
**Pull request by maxtrussell**
The syntax/feature set is much more simplistic than `jq` though. It just takes a dotted set of keys (e.g. "foo.bar.baz") to index with. If one of the given keys is `*` then it will match all elements of the `JsonValue`.

See attached screenshot for more info:
jp

### CODE REVIEW

File: Userland/Utilities/jp.cpp [hunk 1]
```diff
@@ -109,3 +120,41 @@ void print(JsonValue const& value, int spaces_per_indent, int indent, bool use_c
     if (use_color)
         out("\033[0m");
 }
+
+JsonValue query(JsonValue const& value, Vector<StringView>& key_parts, size_t key_index)
+{
+    if (key_index == key_parts.size()) {
+        return value;
+    }
```
Comment (drunderscore) [inline Userland/Utilities/jp.cpp:128]: Single-statement `if`s shouldn't have curly braces

File: Userland/Utilities/jp.cpp [hunk 2]
```diff
@@ -109,3 +120,41 @@ void print(JsonValue const& value, int spaces_per_indent, int indent, bool use_c
...
+
+    JsonValue result {};
+    if (value.is_object()) {
+        result = value.as_object().get(key);
+    } else if (value.is_array()) {
+        auto key_as_index = key.to_int();
+        if (key_as_index.has_value()) {
+            result = value.as_array().at(key_as_index.value());
+        }
```
Comment (drunderscore) [inline Userland/Utilities/jp.cpp:157]: Ditto.

**Review by drunderscore — COMMENTED** [review]
Just these two code smell issues... although I really don't believe a program named `jp` for "JSON Pretty" should have `jq`-like abilities.

File: Userland/Utilities/jp.cpp (hunk 1 above)
Comment (maxtrussell) [reply Userland/Utilities/jp.cpp]: Fixed

File: Userland/Utilities/jp.cpp (hunk 2 above)
Comment (maxtrussell) [reply Userland/Utilities/jp.cpp]: Also fixed

**Comment by maxtrussell** [comment]
> Just these two code smell issues... although I really don't believe a program named `jp` for "JSON Pretty" should have `jq`-like abilities.

@drunderscore That's a fair complaint. I thought of it more as "JSON Print" and printing a subset of the JSON didn't see too off-base to me.

If people feel strongly I could either create a new utility or just close the PR.

If I had to create a new utility, I'd need to:
a. pull out the JSON pretty printing logic to some shared library
b. duplicate it
c. ugly print the json

Given those choices I thought placing it in `jp` was reasonable. And of course, I'm open to alternate ideas if I'm not thinking of something.

File: Userland/Utilities/jp.cpp [hunk 3]
```diff
@@ -109,3 +120,39 @@ void print(JsonValue const& value, int spaces_per_indent, int indent, bool use_c
     if (use_color)
         out("\033[0m");
 }
+
+JsonValue query(JsonValue const& value, Vector<StringView>& key_parts, size_t key_index)
+{
+    if (key_index == key_parts.size())
+        return value;
+    StringView key = key_parts[key_index++];
```
Comment (linusg) [inline Userland/Utilities/jp.cpp:128]: `auto`

File: Userland/Utilities/jp.cpp [hunk 4]
```diff
@@ -109,3 +120,39 @@ void print(JsonValue const& value, int spaces_per_indent, int indent, bool use_c
...
+
+    if (key == "*") {
```
Comment (linusg) [inline Userland/Utilities/jp.cpp:130]: `== "*"sv`

**Review by linusg — CHANGES_REQUESTED** [review]
Second commit needs a little explanation, something along the lines "it no longer just prints JSON"

File: Userland/Utilities/jp.cpp (hunk 3 above)
Comment (maxtrussell) [reply Userland/Utilities/jp.cpp]: Fixed

File: Userland/Utilities/jp.cpp (hunk 4 above)
Comment (maxtrussell) [reply Userland/Utilities/jp.cpp]: Fixed

**Review by linusg — CHANGES_REQUESTED** [review]
You added the changes to the second commit by accident :)

**Comment by maxtrussell** [comment]
> You added the changes to the second commit by accident :)

@linusg Ah dang good catch, just rebased the logic changes into the first commit, so the second is just the rename now.

**Review by linusg — APPROVED** [review]

**Pull request merged**